test/update-many-at-once-upsert.test.ts

```diff
@@ -172,6 +173,68 @@ ava('[UPDATE MANY AT ONCE] Should update one document by _id ObjectID', async (t
 	t.deepEqual(dataUpdatedArray[0].value, 'update', 'value is updated');
 });
 
+ava('[UPDATE MANY AT ONCE] Should call mapAfterLockFieldsApplied', async (t: Assertions) => {
+	const collectionName = `test-${Date.now()}`;
+	const mongoClient = new MongoClient(collectionName, SampleType, SampleType, {
+		lockFields: {
+			arrayWithReferences: {
+				externalReferences: 'value',
+			},
+			excludedFields: ['sku', 'status'],
+		},
+	});
+
+	const insertedValue = await mongoClient.insertOne(
+		{
+			externalReferences: [
+				{
+					value: 'ext1',
+				},
+			],
+			sku: 'sku-1',
+			value: '0',
+		},
+		'test',
+	);
+	t.is(insertedValue.value, '0', 'value inserted');
+	t.is(insertedValue.status, undefined, 'status is undefined');
+	t.is(insertedValue.objectInfos.lockFields.length, 2, 'should lock new fields');
+
+	const newEntity: SampleType = {
+		sku: 'sku-1',
+		externalReferences: [
+			{
+				value: 'ext1',
+			},
+		],
+		value: '1',
+	};
+
+	const updateResult = await (
+		await mongoClient.updateManyAtOnce([newEntity], 'userId', {
```

Comment:
I think you should add another case for the upsert to test the hook on insertion.
This will allow to test line client.ts:1630

Reply:
You can have a look here : https://app.codecov.io/gh/neo9/n9-mongo-client/pull/57

Reply:
done!